src/Helpers/LocalMachineHelper.php

```diff
@@ -204,12 +204,16 @@ protected function fixFilename($filename)
      */
     protected function getProcess($cmd)
     {
+        $process = null;
         if (is_string($cmd)) {
-            $cmd = [ $cmd ];
+            $process = Process::fromShellCommandline($cmd);
+        } elseif (is_array($cmd)) {
+            $process = new Process($cmd);
+        }
+        if ($process) {
```

Comment:
The existing method would never return a `null` result. Maybe we should throw an exception on bad inputs to better match the old contract.

Reply:
@namespacebrian ^^^

Reply:
No, I would throw an invalid argument exception. I think that allowing $cmd to be an array would be a helpful way to allow the code to transition to the modern calling semantics. If we were going to type $cmd to anything, I'd say force it to be an array and fix all of the callers. But that's probably not a good thing to do right now, as plugins might use this method (via exec()), and we don't want 3.x to have such a large break in it.

Reply:
Agree. I updated the code to throw the exception and updated parameter documentation.